astlib/migrate_501_500.ml

```diff
@@ -227,16 +227,72 @@ and copy_value_binding :
      } ->
   let pvb_pat = copy_pattern pvb_pat and pvb_expr = copy_expression pvb_expr in
   let pvb_pat, pvb_expr =
-    match pvb_constraint with
-    | None -> (pvb_pat, pvb_expr)
-    | Some { locally_abstract_univars; typ } ->
+    match (pvb_constraint, pvb_pat) with
+    | ( Some (Pvc_constraint { locally_abstract_univars; typ }),
```

Comment:
This case perform a supplementary deep copy, it is better to make sure that it only happens when `locally_abstract_univars` is not empty.

Reply:
I had missed that this case also cover the special encoding `let ident: typ ` in general. It is probably thus simpler to avoid calling `varify_constructors` when `locally_abstract_univars` are empty.